### PR TITLE
CPLAT-6510 Finalize UiComponent2 - part 1 of X

### PR DESCRIPTION
**Pull request by greglittlefield-wf**
# Depends on https://github.com/cleandart/react-dart/pull/195

## Motivation
Component2 was implemented via a large number of PRs, and has some parts that need to be cleaned up before release.

## Changes
- Update imports to use new top-level entrypoints and not src/ files
- Stub in a custom UiComponent bridge for use in prop validation PR
- Add automatic typing to `UiStatefulComponent.setStateWithUpdater` callbacks:
    ```dart
    setStateWithUpdater((prevState, props) {
      // prevState is statically typed as TState
      // props is statically typed as TProps
    });
    ```

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

Please review: @aaronlademann-wf @kealjones-wk @joebingham-wk @sydneyjodon-wk 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_

[contributing-review-types]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: pubspec.yaml [hunk 1]
```diff
@@ -42,4 +42,4 @@ dependency_overrides:
   react:
     git:
       url: https://github.com/cleandart/react-dart.git
-      ref: 5.1.0-wip
+      ref: finalize-component2-api/dev
```
Comment (greglittlefield-wf) [inline pubspec.yaml]: Remove before merging

**Comment by kealjones-wk** [comment]
+10

**Comment by aaronlademann-wf** [comment]
+1

**Pull request merged**